### PR TITLE
Multiple mech related fixes

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Fixes landmines not triggering over mechs due to a runtime
Fixes SWING_ATTACK not attacking plants in general
Fixes mech blades having the wrong description , and their ability to embed into the user.
## Why It's Good For The Game
Fixes
## Testing
Ran on local

## Changelog
:cl:
fix:Fixes landmines not triggering over mechs due to a runtime
fix:Fixes SWING_ATTACK not attacking plants in general
fix:Fixes mech blades having the wrong description , and their ability to embed into the user.

/:cl:

### CODE REVIEW

**Comment by Firefox13** [comment]
> Ran on local

The testing section is not to say "yeah I tested it" but to describe how you actually tested it. Please you made like 4 hotfixes for 2 PRs even if it's a simple variable change, do **test** it and describe **how** you tested it.

**Comment by MLGTASTICa** [comment]
> > Ran on local
> 
> The testing section is not to say "yeah I tested it" but to describe how you actually tested it. Please you made like 4 hotfixes for 2 PRs even if it's a simple variable change, do **test** it and describe **how** you tested it.

I tested it by running it on local , walking over a landmine with a mech , swinging at plants and examining the blade.

File: code/game/objects/items/weapons/mine.dm [hunk 1]
```diff
@@ -199,10 +199,18 @@
 			visible_message(SPAN_DANGER("\The [src]'s triggering mechanism is disrupted by the slope and does not go off."))
 			return ..()
 		if(isliving(AM))
+
 			if(excelsior)
-				for(var/datum/antagonist/A in AM.mind.antagonist)
-					if(A.id == ROLE_EXCELSIOR_REV)
-						return
+				if(ismech(AM))
+					/// if atleast one of the people inside if an excel.
```
Comment (hyperioo) [inline code/game/objects/items/weapons/mine.dm:205]:
```suggestion
					/// if at least one of the people inside is an excel.
```

File: code/game/objects/items/weapons/tools/_tools.dm [hunk 2]
```diff
@@ -528,7 +528,8 @@
 		if(T && T.degradation)
 			failtypes["damage"] = 2.5
 
-	if(user)
+	// You can only fail with tools you are holding
+	if(user && T.loc == user)
```
Comment (hyperioo) [inline code/game/objects/items/weapons/tools/_tools.dm:532]: And a quick fix to tool code am I right?

**Pull request merged**